### PR TITLE
Make GC stress tests cross platform

### PR DESCRIPTION
**Pull request by ludovic-henry**
Use .NET provided path facilities, instead of using hardcoded values, such as path separator.

### CODE REVIEW

**Comment by mmitche** [comment]
@dotnet-bot test linux

**Comment by mmitche** [comment]
Ignore any failures coming from those runs, this is just for testing

File: tests/src/GC/Stress/Framework/ReliabilityFramework.cs [hunk 1]
```diff
@@ -636,9 +636,9 @@ public void GetPerfStats(ref float memVal, ref float cpuVal, ref float pagesVal,
         if (myProcessName == null)
         {
             myProcessName = System.Windows.Forms.Application.ExecutablePath;
-            if (myProcessName.LastIndexOf("\\") != -1)
+            if (myProcessName.LastIndexOf(Path.SeparatorPath) != -1)
             {
-                myProcessName = myProcessName.Substring(myProcessName.LastIndexOf("\\") + 1);
+                myProcessName = myProcessName.Substring(myProcessName.LastIndexOf(Path.SeparatorPath) + 1);
```
Comment (akoeplinger) [inline tests/src/GC/Stress/Framework/ReliabilityFramework.cs]: should be `Path.PathSeparator` here and above
(talked with Ludovic about this already, just commenting here so it's not forgot)

**Comment by jkotas** [comment]
LGTM

**Comment by ludovic-henry** [comment]
Thank you very much! :)

**Pull request merged**